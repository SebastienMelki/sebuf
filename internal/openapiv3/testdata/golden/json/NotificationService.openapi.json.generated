{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"Request":{"description":"Generic request message","properties":{"data":{"description":"Request data","type":"string"},"id":{"description":"Request ID","type":"string"}},"type":"object"},"Response":{"description":"Generic response message","properties":{"data":{"description":"Response data","type":"string"},"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"NotificationService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/notifications/email/send":{"post":{"description":"Send email notification","operationId":"SendEmail","parameters":[{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendEmail","tags":["NotificationService"]}},"/api/v1/notifications/push/send":{"post":{"description":"Send push notification","operationId":"SendPush","parameters":[{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendPush","tags":["NotificationService"]}},"/api/v1/notifications/sms/send":{"post":{"description":"Send SMS notification","operationId":"SendSMS","parameters":[{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}},{"description":"SMS provider to use","in":"header","name":"X-SMS-Provider","required":false,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendSMS","tags":["NotificationService"]}}}}