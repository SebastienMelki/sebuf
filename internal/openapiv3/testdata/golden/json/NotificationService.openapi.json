{"components":{"schemas":{"Request":{"description":"Generic request message","properties":{"data":{"description":"Request data","type":"string"},"id":{"description":"Request ID","type":"string"}},"type":"object"},"Response":{"description":"Generic response message","properties":{"data":{"description":"Response data","type":"string"},"message":{"description":"Response message","type":"string"},"success":{"description":"Success indicator","type":"boolean"}},"type":"object"},"User":{"description":"User message used by multiple services","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"},"role":{"description":"User role","type":"string"}},"type":"object"}}},"info":{"title":"NotificationService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/notifications/email/send":{"post":{"description":"Send email notification","operationId":"SendEmail","parameters":[{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendEmail","tags":["NotificationService"]}},"/api/v1/notifications/push/send":{"post":{"description":"Send push notification","operationId":"SendPush","parameters":[{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendPush","tags":["NotificationService"]}},"/api/v1/notifications/sms/send":{"post":{"description":"Send SMS notification","operationId":"SendSMS","parameters":[{"description":"SMS provider to use","in":"header","name":"X-SMS-Provider","required":false,"schema":{"type":"string"}},{"description":"Notification service API key","in":"header","name":"X-Notification-Key","required":true,"schema":{"type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Response"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SendSMS","tags":["NotificationService"]}}}}